server/engine/src/main/java/org/eclipse/lsp/cobol/cfg/CFASTBuilderImpl.java

```diff
@@ -107,7 +107,7 @@ private void traverse(CFASTNode parent, Node node) {
       node.getChildren().forEach(child -> traverse(parent, child));
     } else if (node instanceof PerformNode) {
       if (((PerformNode) node).isInline()) {
-        addChild(parent, new CFASTNode(CFASTNodeType.INLINE_PERFORM.getValue(), convertLocation(node)));
+        addChild(parent, new InlinePerform(convertLocation(node), ((PerformNode) node).isUntilExit()));
         node.getChildren().forEach(child -> traverse(parent, child));
         addChild(parent, new CFASTNode(CFASTNodeType.END_INLINE_PERFORM.getValue(), convertLocation(node)));
       } else {
```

Comment:
The UNTIL EXIT clause seems to be available also for the normal perform.
Do you want to address all of it in this PR?

Reply:
I have added this logic to the current PR, thanks!